http.go

```diff
@@ -0,0 +1,36 @@
+package server
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+// writeError writes err to w in JSON format. If err is
+// internal, obfuscates message and logs it the stdout.
+func writeError(w http.ResponseWriter, err error) {
+	code := errorCode(err)
+	msg := errorMessage(err)
+
+	if code >= http.StatusInternalServerError {
+		// Obfuscate message.
+		msg = http.StatusText(code)
+	}
+	// TODO verbose mode and/or logging to file.
+	log.Println(err.Error())
+
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+
+	writeJSON(w, httpError{Message: msg}, code)
+}
+
+// writeJSON writes data to w in JSON format and sets the response satus code.
+func writeJSON(w http.ResponseWriter, data interface{}, code int) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+
+	if err := json.NewEncoder(w).Encode(data); err != nil {
+		writeError(w, &ErrInternal)
```

Comment:
It was there in the [previous implementation](https://github.com/benchttp/server/blob/main/http/response.go#L15-L33) too (and all our other projects) 😛

Reply:
Never noticed it before 🤦‍♂️ 
Let's keep it for further improvement if it's annoying to fix, but we really need to fix it else it'll be in our next project as well 😅

Reply:
Well, if we want to send an error response in JSON format we have to call `json.NewEncoder(w).Encode(data)` and thus handle its failure ...

What we could do is:
- `writeError` calls `writeJSON`
- if `writeJSON` fails from encoding JSON and `data` reflects to an error, write it in plain text

Is it worth ?

Reply:
Seems good, I'd do even simpler and send the raw 500 as soon as the JSON encoding fails without further checks.

Honestly the `writeError` call is only a problem if `writeError` itself sends an unmarshalable structure which seems very unlikely. But given it's easy to fix, I'd rather make the 🔄 impossible by design rather than extremely unlikely.

But again it can be done as an improvement later to me!